File: 2020/02/2.js
// 2020 Day 2 Part 2
const fs = require('fs');
let input = fs.readFileSync('input.txt', 'utf-8').split("\n");
let count = 0;

input = input.map(x => {
  return x.split(/[\- ]|\: /g);
});

for (key of input) {
  let pos1 = parseInt(key[0]);
  let pos2 = parseInt(key[1]);
  let char = key[2];
  let password = key[3];

  if ((password[pos1-1] == char && password[pos2-1] != char) || 
  (password[pos1-1] != char && password[pos2-1] == char)) {
    count++;
  }
}

console.log(count);

File: 2017/01/2.swift
import Foundation

var input = try String(contentsOfFile: "./input.txt")
  .split(separator: "")
  .map { Int($0)! }

let count = input.count
input += input

var sum = 0;
for i in 0..<count {
  if input[i] == input[i+count/2] { sum += input[i] }
}

print(sum)
